packages/quill/src/blots/embed.ts

```diff
@@ -78,16 +78,18 @@ class Embed extends EmbedBlot {
   }
 
   update(mutations: MutationRecord[], context: Record<string, unknown>) {
-    mutations.forEach((mutation) => {
-      if (
+    const mutation = mutations.find((mutation) => {
+      return (
         mutation.type === 'characterData' &&
         (mutation.target === this.leftGuard ||
           mutation.target === this.rightGuard)
-      ) {
-        const range = this.restore(mutation.target as Text);
-        if (range) context.range = range;
-      }
+      );
     });
+
+    if (mutation) {
+      const range = this.restore(mutation.target as Text);
```

Comment:
here is what `this.restore` does:
- get text from `mutation.target.data`,
- calculate range by text
- reset `mutation.target.data` to empty

so only the first execution of `this.restore` is effective
the next execution will get empty text from ``mutation.target.data`, and return wrong range, which is the cause of the issue https://github.com/quilljs/quill/issues/4046